.github/workflows/openhands-resolver.yml

```diff
@@ -178,7 +210,7 @@ jobs:
           fi
 
           echo "MAX_ITERATIONS=${{ inputs.max_iterations || 50 }}" >> $GITHUB_ENV
-          echo "SANDBOX_ENV_GITHUB_TOKEN=${{ secrets.PAT_TOKEN || github.token }}" >> $GITHUB_ENV
+          echo "SANDBOX_ENV_GITHUB_TOKEN=${{ steps.determine-auth-token.outputs.EFFECTIVE_TOKEN }}" >> $GITHUB_ENV
```

Comment:
It's odd that we're access `EFFECTIVE_TOKEN` through the step's output. I would imagine that wouldn't be allowed by workflows as that would leak the secret. 

Could we change this so that its being accessed from the env instead? Same for all other places effective token is being used

Reply:
Updated. Though I _think_ it would be fine within the same job. It definitely get's blocked if you try to pass secrets between jobs. There's a whole thread on it: https://github.com/actions/create-github-app-token/issues/66